src/repository/default.rs

```diff
@@ -439,6 +439,51 @@ impl Repository for $repo {
         Ok(functions)
     }
 
+
+    async fn drop_function(
+        &self,
+        database_id: DatabaseId,
+        if_exists: bool,
+        func_names: &[String],
+    ) -> Result<(), Error> {
+
+        for function_name in func_names.iter() {
+            // Construct the SQL DELETE statement
+            let query =
+                format!(
+                    r#"
+                    DELETE FROM "function"
+                    WHERE database_id = $1 AND name = $2
+                    RETURNING id;
+                    "#,
+                );
+
+            // Execute the SQL DELETE query 1x per function
+            let query = sqlx::query(&query)
+                .bind(database_id)
+                .bind(function_name);
+
+            println!("database_id: {}, function_name: {}", database_id, function_name);
+
+            if if_exists {
+                // If user provided IF EXISTS, tolerate when no function found
+                query
+                    .execute(&self.executor)
+                    .await
+                    .map_err($repo::interpret_error)?;
+            } else {
+                // Otherwise if no function found, we return error
+                query
+                    .fetch_one(&self.executor)
+                    .await
+                    .map_err($repo::interpret_error)?;
+            }
+
+        }
```

Comment:
If you go for handling/converting `Err(RepositoryError::SqlxError(sqlx::error::Error::RowNotFound))` in the catalog, then you can simplify this to one expression (using `fetch_one`), and there's no need to pass `if_exists` down here.

Reply:
Done 👍